Index: agavi/src/validator/AgaviValidator.class.php
===================================================================
--- agavi/src/validator/AgaviValidator.class.php	(revision 1293)
+++ agavi/src/validator/AgaviValidator.class.php	(working copy)
@@ -489,12 +489,20 @@
 			$this->affectedArguments = $affectedArguments;
 		}
 
+		$translationParameters = $this->getParameters();
+
+		if (count($this->affectedArguments) > 1) {
+			$translationParameters['arguments'] = implode(', ', $this->affectedArguments);
+		} elseif(isset($this->affectedArguments[0])) {
+			$translationParameters['argument'] = $this->affectedArguments[0];
+		}
+
 		$error = $this->getErrorMessage($index);
 
 		if($this->hasParameter('translation_domain')) {
-			$error = $this->getContext()->getTranslationManager()->_($error, $this->getParameter('translation_domain'));
+			$error = $this->getContext()->getTranslationManager()->_($error, $this->getParameter('translation_domain'), null, $translationParameters);
 		} else {
-			$error = $this->getContext()->getTranslationManager()->_($error);
+			$error = $this->getContext()->getTranslationManager()->_($error, null, null, $translationParameters);
 		}
 
 		if(!$this->incident) {
