Index: agavi/src/routing/AgaviWebRouting.class.php
===================================================================
--- agavi/src/routing/AgaviWebRouting.class.php	(revision 969)
+++ agavi/src/routing/AgaviWebRouting.class.php	(working copy)
@@ -427,7 +427,7 @@
 		}
 
 		if($options['fragment'] !== null) {
-			$retval .= '#' . $options['fragment'];
+			$retval .= '#' . $this->escapeOutputParameter($options['fragment']);
 		}
 
 		return $retval;
@@ -450,4 +450,4 @@
 
 }
 
-?>
\ No newline at end of file
+?>
